arwen/contexts/metering.go

```diff
@@ -194,8 +198,25 @@ func (context *meteringContext) deductInitialGas(
 ) error {
 	input := context.host.Runtime().GetVMInput()
 	codeLength := uint64(len(code))
-	codeCost := codeLength * costPerByte
-	initialCost := baseCost + codeCost
+	codeCost, err := math.MulUint64WithErr(codeLength, costPerByte)
+	if err != nil {
+		log.Error("deductInitialGas overflow",
+			"codeLength = ", codeLength,
+			"costPerByte = ", costPerByte,
+		)
+
+		return arwen.ErrNotEnoughGas
+	}
+
+	initialCost, err := math.AddUint64WithErr(baseCost, codeCost)
```

Comment:
Same as above, `math.AddUint64()` without `Err` would be cleaner, yet equivalent due to the condition at 221.

Reply:
Done.